tensorflow_datasets/core/download/kaggle.py

```diff
@@ -49,156 +49,92 @@
-      dataset_name, filename = filename.split("/", 1)
-      competition_name = "%s/%s" % (competition_name, dataset_name)
-    return cls(competition_name, filename)
-
-  @staticmethod
-  def is_kaggle_url(url):
-    return url.startswith(KaggleFile._URL_PREFIX)
-
-  def to_url(self):
-    return "%s%s/%s/%s" % (self._URL_PREFIX,
-                           self.type.prefix,
-                           self._competition_name,
-                           self._filename)
-
-
-class KaggleCompetitionDownloader(object):
-  """Downloader for a Kaggle competition.
-
-  Usage:
-  You can download with dataset or competition name like `zillow/zecon`
-  or `titanic`.
-
-  ```
-  downloader = KaggleCompetitionDownloader(competition_name)
-  for fname in downloader.competition_files:
-    downloader.download_file(fname, make_file_output_path(fname))
-  ```
-  """
-
-  def __init__(self, competition_name):
-    self._competition_name = competition_name
-    self._kaggle_type = _get_kaggle_type(self._competition_name)
-
-  @utils.memoized_property
-  def competition_files(self):
-    """List of competition files."""
-    command = [
-        "kaggle",
-        self._kaggle_type.download_cmd,
-        "files",
-        "-v",
-        self._competition_name,
-    ]
-    output = _run_kaggle_command(command, self._competition_name)
-    return sorted([
-        line.split(",")[0] for line in output.split("\n")[1:] if line
-    ])
-
-  @utils.memoized_property
-  def competition_urls(self):
-    """Returns 'kaggle://' urls."""
-    return [
-        KaggleFile(self._competition_name, fname).to_url()
-        for fname in self.competition_files  # pylint: disable=not-an-iterable
-    ]
-
-  def download_file(self, fname, output_dir):
-    """Downloads competition file to output_dir."""
-    if fname not in self.competition_files:  # pylint: disable=unsupported-membership-test
-      raise ValueError("%s is not one of the competition's "
-                       "files: %s" % (fname, self.competition_files))
-    command = [
-        "kaggle",
-        self._kaggle_type.download_cmd,
-        "download",
-        "--file",
-        fname,
-        "--path",
-        output_dir,
-        self._kaggle_type.dl_flag,
-        self._competition_name
-    ]
-    if self._kaggle_type.extra_flag:
-      command.append(self._kaggle_type.extra_flag)
-    _run_kaggle_command(command, self._competition_name)
-    # kaggle silently compresses some files to '.zip` files.
-    # TODO(tfds): use --unzip once supported by kaggle
-    # (https://github.com/Kaggle/kaggle-api/issues/9)
-    fpath = os.path.join(output_dir, fname + ".zip")
-    if zipfile.is_zipfile(fpath):
-      e = extractor.get_extractor()
-      with e.tqdm():
-        e.extract(fpath, resource.ExtractMethod.ZIP, output_dir).get()
-    return os.path.join(output_dir, fname)
+def is_kaggle_url(url):
+  return url.startswith("kaggle.com")
+
+
+def kaggle_dir_name(competition_or_dataset):
```

Comment:
For new functions, we should add typing annotations (`def _kaggle_dir_name(competition_or_dataset: str) -> str`)

Reply:
Sounds good. I was going to update docstrings, fix lint errors, etc. in a separate PR